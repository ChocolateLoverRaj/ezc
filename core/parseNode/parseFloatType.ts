import CoreKeyWord from '../CoreKeyWord'
import CoreTokenType from '../CoreTokenType'
import reverseMap from '../../util/reverseMap/reverseMap'
import CoreNodesWithData from './CoreNodesWithData'
import CoreNodeType from './CoreNodeType'
import floatTypeToKeyWord from './floatTypeToKeyWord'
import TryParseNode from './TryParseNode'
import checkToken from './checkToken'
import EnumItem from '../EnumItem'
import TryParseNodeResult from './tryParseNodeResult/TryParseNodeResult'

const parseFloatType: TryParseNode<CoreNodesWithData[CoreNodeType.FLOAT_TYPE]> = async stream => {
  const type = {
    enum: CoreNodeType,
    id: CoreNodeType.FLOAT_TYPE
  }

  const message = 'Expected float type'
  const result = await checkToken(
    stream[Symbol.asyncIterator](),
    type,
    0,
    message,
    { enum: CoreTokenType, id: CoreTokenType.KEY_WORD })
  if (!result.success) return result
  const enumItem = result.result as EnumItem
  const error: TryParseNodeResult<CoreNodesWithData[CoreNodeType.FLOAT_TYPE]> = {
    success: false,
    result: {
      type,
      index: 0,
      message,
      subAttempts: undefined
    }
  }
  if (enumItem.enum !== CoreKeyWord) return error
  const floatType = reverseMap(floatTypeToKeyWord).get(enumItem.id)
  if (floatType === undefined) return error

  return {
    success: true,
    result: {
      node: {
        type,
        data: floatType
      },
      length: 1
    }
  }
}

export default parseFloatType
