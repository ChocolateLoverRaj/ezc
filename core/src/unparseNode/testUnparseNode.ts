import TryParseNode from '../parseNode/TryParseNode'
import EnumItemWithData from '../EnumItemWithData'
import UnparsedPartType from './UnparsedPartType'
import unparsedNodeToString from '../unparsedNodeToString/unparsedNodeToString'
import coreInput from '../unparsedNodeToString/coreInput'
import parseAllTokens from '../tryGetToken/parseAllTokens'
import coreTryers from '../tryGetToken/coreTryers'
import arrayToAsyncIterable from 'util/dist/arrayToAsyncIterable/arrayToAsyncIterable.js'

const testUnparseNode = async (
  parseNode: TryParseNode<EnumItemWithData>,
  code: string
): Promise<void> => {
  expect(unparsedNodeToString(coreInput)([{
    type: UnparsedPartType.NODE,
    data: ((await parseNode(
      parseAllTokens(coreTryers)(
        arrayToAsyncIterable([code])) as any))?.result as any).node as EnumItemWithData
  }])).toMatchSnapshot()
}

export default testUnparseNode
