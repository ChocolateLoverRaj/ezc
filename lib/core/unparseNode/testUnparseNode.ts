import TryParseNode from '../parseNode/TryParseNode'
import EnumItemWithData from '../EnumItemWithData'
import UnparsedPartType from './UnparsedPartType'
import unparsedNodeToString from '../unparsedNodeToString/unparsedNodeToString'
import coreInput from '../unparsedNodeToString/coreInput'
import parseAllTokens from '../tryGetToken/parseAllTokens'
import coreTryers from '../tryGetToken/coreTryers'
import arrayToAsyncIterable from '../arrayToAsyncIterable'

const testUnparseNode = async (
  parseNode: TryParseNode<EnumItemWithData>,
  code: string
): Promise<void> => {
  expect(unparsedNodeToString(coreInput)([{
    type: UnparsedPartType.NODE,
    data: (await parseNode(
      parseAllTokens(coreTryers)(
        arrayToAsyncIterable([code])) as any))?.node as EnumItemWithData
  }])).toMatchSnapshot()
}

export default testUnparseNode
