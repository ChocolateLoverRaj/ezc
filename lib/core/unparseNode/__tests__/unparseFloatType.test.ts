import parseFloatType from '../../parseNode/parseFloatType'
import testUnparseNode from '../testUnparseNode'

test('double', async () => {
  await testUnparseNode(parseFloatType, 'double')
})
