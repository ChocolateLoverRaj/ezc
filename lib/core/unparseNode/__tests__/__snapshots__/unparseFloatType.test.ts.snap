// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`double 1`] = `"double"`;
